### PR TITLE
Issue60 12 FA Compliant & Remove use of common utils

### PR DESCRIPTION
**Pull request by emilioward**
This PR addresses 2 cards to remove the use common utils and change the config so it gets configured from its environment (https://trello.com/c/aL4Mm2Cr) & (https://trello.com/c/Wd0IK72m)

### CODE REVIEW

**Comment by emilioward** [comment]
[Make collection Instrument 12FA compliant](https://trello.com/c/aL4Mm2Cr/8-make-collection-instrument-12fa-compliant)

**Comment by emilioward** [comment]
[Remove use of common-utils](https://trello.com/c/10pkGUX9/3-remove-use-of-common-utils)

File: app.py [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
+# # This is a duplicate of run.py, with minor modifications to support gunicorn execution.
+#
+# logger = structlog.get_logger()
+#
+# config_path = 'config.py'
+# config = os.environ['APP_SETTINGS'] = 'Config'
+#
+# app = create_app()
+# configure_logger(app.config)
+# logger.debug("Created Flask application.")
```
Comment (gemmai95) [inline app.py]: Why is this all being commented rather than removed?

File: application/controllers/rabbitmq_submitter.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 from pika import BlockingConnection, BasicProperties, URLParameters
 from pika.exceptions import AMQPError
-from ras_common_utils.ras_error.ras_error import RasError
+from application.exceptions import RasError
```
Comment (gemmai95) [inline application/controllers/rabbitmq_submitter.py]: Don't think this adheres to PEP8 import standards. If you are importing from the application itself this should be a third group of imports at the end of the rest of imports.

File: application/controllers/service_helper.py [hunk 3]
```diff
@@ -1,10 +1,12 @@
 import requests
+import logging
+import structlog
 
 from flask import current_app
-from ras_common_utils.ras_error.ras_error import RasError
-from structlog import get_logger
+from application.exceptions import RasError
```
Comment (gemmai95) [inline application/controllers/service_helper.py]: Could you break these up by adding a space, to make clear it is a separate import group?

File: application/controllers/session_decorator.py [hunk 4]
```diff
@@ -1,10 +1,12 @@
-from functools import wraps
+import logging
+import structlog
 
+from functools import wraps
 from flask import current_app
-from ras_common_utils.ras_error.ras_error import RasDatabaseError, RasError
-from structlog import get_logger
+from application.exceptions import RasDatabaseError, RasError
```
Comment (gemmai95) [inline application/controllers/session_decorator.py]: Same again, please.

File: application/controllers/survey_response.py [hunk 5]
```diff
@@ -1,16 +1,17 @@
 import os
 import time
 import uuid
+import logging
```
Comment (gemmai95) [inline application/controllers/survey_response.py]: Alphabetical Imports

File: application/error_handlers.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import logging
+import structlog
+
+from application.exceptions import RasError
```
Comment (gemmai95) [inline application/error_handlers.py]: Should be at the bottom in a separate import group

File: application/logger_config.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import logging
+import os
+
+from structlog import configure
+from structlog.stdlib import add_log_level, filter_by_level
+from structlog.processors import JSONRenderer, TimeStamper
```
Comment (gemmai95) [inline application/logger_config.py]: Alphabetical imports

File: application/models/models.py [hunk 8]
```diff
@@ -1,12 +1,14 @@
 from .associations import instrument_exercise_table, instrument_business_table
 from datetime import datetime
-from ras_common_utils.ras_database.base import Base
-from ras_common_utils.ras_database.guid import GUID
+from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy import Column, Integer, ForeignKey
 from sqlalchemy.orm import relationship
 from sqlalchemy.types import TIMESTAMP, LargeBinary, Enum, String
+from application.models import GUID
```
Comment (gemmai95) [inline application/models/models.py]: Needs to be in it's own import group

File: config.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    UPLOAD_FILE_XLS = os.getenv('UPLOAD_FILE_XLS', 'xls')
+    UPLOAD_FILE_XLSX = os.getenv('UPLOAD_FILE_XLSX', 'xlsx')
+    UPLOAD_FILE_EXTENSIONS = (UPLOAD_FILE_XLS, UPLOAD_FILE_XLSX)
+
+    MAX_UPLOAD_FILE_NAME_LENGTH = os.getenv('MAX_UPLOAD_FILE_NAME_LENGTH', 50)
+
+    RAS_COLLECTION_INSTRUMENT_DATABASE_SCHEMA = os.getenv('RAS_COLLECTION_INSTRUMENT_DATABASE_SCHEMA', 'ras_ci')
+    RAS_COLLECTION_INSTRUMENT_DATABASE_URI = os.getenv('RAS_COLLECTION_INSTRUMENT_DATABASE_URI', 'sqlite:///ras-ci')
+
+    CASE_SERVICE_PROTOCOL = os.getenv('CASE_SERVICE_PROTOCOL', 'http')
```
Comment (gemmai95) [inline config.py]: Are we still using this method of config to include the protocol separately?

File: run.py [hunk 10]
```diff
@@ -1,56 +1,75 @@
+import os
+import logging
+import structlog
+from application.logger_config import logger_initial_config
```
Comment (gemmai95) [inline run.py]: Same again with in app imports

File: tests/controllers/test_collection_instrument.py [hunk 11]
```diff
@@ -3,7 +3,7 @@
 from application.controllers.collection_instrument import CollectionInstrument
 from application.controllers.session_decorator import with_db_session
 from application.models.models import ExerciseModel, InstrumentModel, BusinessModel, SurveyModel
-from ras_common_utils.ras_error.ras_error import RasDatabaseError, RasError
+from application.exceptions import RasDatabaseError, RasError
```
Comment (gemmai95) [inline tests/controllers/test_collection_instrument.py]: In app imports should be separate group

File: tests/controllers/test_helper.py [hunk 12]
```diff
@@ -3,7 +3,7 @@
 from application.controllers.helper import convert_string_to_bytes_base64, validate_uuid, \
     convert_file_object_to_string_base64, to_str, to_bytes, is_valid_file_extension, is_valid_file_name_length
 from werkzeug.datastructures import FileStorage
-from ras_common_utils.ras_error.ras_error import RasError
+from application.exceptions import RasError
```
Comment (gemmai95) [inline tests/controllers/test_helper.py]: Same again

File: tests/controllers/test_rabbitmq_submitter.py [hunk 13]
```diff
@@ -2,7 +2,7 @@
 from unittest.mock import patch
 
 from pika.exceptions import AMQPError
-from ras_common_utils.ras_error.ras_error import RasError
+from application.exceptions import RasError
```
Comment (gemmai95) [inline tests/controllers/test_rabbitmq_submitter.py]: Same again

File: tests/controllers/test_service_helper.py [hunk 14]
```diff
@@ -1,5 +1,5 @@
 from application.controllers.service_helper import service_request
-from ras_common_utils.ras_error.ras_error import RasError
+from application.exceptions import RasError
```
Comment (gemmai95) [inline tests/controllers/test_service_helper.py]: Same again

File: tests/test_client.py [hunk 15]
```diff
@@ -1,19 +1,23 @@
+import logging
+import structlog
+import os
```
Comment (gemmai95) [inline tests/test_client.py]: Alphabetical imports

File: Pipfile [hunk 16]
```diff
@@ -41,3 +42,5 @@ flask-testing = "*"
 pytest = "*"
```
Comment (jcox-dev) [inline Pipfile]: Before you start pulling your hair out, add: `flask-testing = "==0.6.2"`

File: application/controllers/collection_instrument.py [hunk 17]
```diff
@@ -1,15 +1,17 @@
...
+
 from application.controllers.cryptographer import Cryptographer
 from application.controllers.helper import validate_uuid
 from application.controllers.service_helper import service_request
 from application.controllers.session_decorator import with_db_session
 from application.controllers.sql_queries import query_business_by_ru, query_exercise_by_id, query_instrument, \
     query_instrument_by_id, query_survey_by_id
+from application.exceptions import RasError
 from application.models.models import BusinessModel, ClassificationModel, ExerciseModel, InstrumentModel, SurveyModel
 from json import loads
```
Comment (gemmai95) [inline application/controllers/collection_instrument.py]: Can you move this?

File: application/controllers/helper.py [hunk 18]
```diff
@@ -1,6 +1,6 @@
 import base64
 
-from ras_common_utils.ras_error.ras_error import RasError
+from application.exceptions import RasError
 from uuid import UUID
```
Comment (gemmai95) [inline application/controllers/helper.py]: And this

File: application/views/collection_instrument_view.py [hunk 19]
```diff
@@ -1,10 +1,11 @@
+import logging
 from application.controllers.basic_auth import auth
```
Comment (gemmai95) [inline application/views/collection_instrument_view.py]: Break these into groups?

File: application/views/survey_responses_view.py [hunk 20]
```diff
@@ -1,12 +1,14 @@
+import logging
+import os
+
 from application.controllers.basic_auth import auth
```
Comment (gemmai95) [inline application/views/survey_responses_view.py]: These too

File: app.py [hunk 21]
```diff
@@ -1,20 +1,30 @@
...
+logger = structlog.wrap_logger(logging.getLogger(__name__))
 
-logger = structlog.get_logger()
+app = create_app()
+with open(app.config['COLLECTION_INSTRUMENT_SCHEMA']) as io:
+    app.config['COLLECTION_INSTRUMENT_SCHEMA'] = loads(io.read())
 
-config_path = 'config/config.yaml'
-config = ras_config.from_yaml_file(config_path)
+logger_initial_config(service_name='ras-collection-instrument', log_level=app.config['LOG_LEVEL'])
```
Comment (jcox-dev) [inline app.py]: LOGGING_LEVEL in config.

**Pull request merged**